bin/wiki_entity_linking/entity_linker_evaluation.py

```diff
@@ -0,0 +1,247 @@
+
+                    best_index = scores.index(max(scores))
+                    best_candidate = candidates[best_index].entity_
+                    random_candidate = random.choice(candidates).entity_
+
+                random_correct_d[label] += (gold_entity == random_candidate)
+                random_incorrect_d[label] += (not gold_entity == random_candidate)
+
+                random_true_positives += (gold_entity == random_candidate)
+                random_false_negatives += (not gold_entity == random_candidate)
+                if candidates:
+                    random_false_positives += (not gold_entity == random_candidate)
+
+                prior_correct_d[label] += (gold_entity == best_candidate)
+                prior_incorrect_d[label] += (not gold_entity == best_candidate)
+
+                prior_true_positives += (gold_entity == best_candidate)
+                prior_false_negatives += (not gold_entity == best_candidate)
+                if candidates:
+                    prior_false_positives += (not gold_entity == best_candidate)
+
+                oracle_correct_d[label] += (gold_entity == oracle_candidate)
+                oracle_incorrect_d[label] += (not gold_entity == oracle_candidate)
+
+                oracle_true_positives += (gold_entity == oracle_candidate)
+                oracle_false_negatives += (not gold_entity == oracle_candidate)
+                if candidates:
+                    oracle_false_positives += (not gold_entity == oracle_candidate)
+
+    return BaselineResults(
+        counts_d,
+        _calculate_recall(random_true_positives, random_false_negatives),
+        _calculate_precision(random_true_positives, random_false_positives),
+        calculate_acc(random_correct_d, random_incorrect_d)[1],
+        _calculate_recall(prior_true_positives, prior_false_negatives),
+        _calculate_precision(prior_true_positives, prior_false_positives),
+        calculate_acc(prior_correct_d, prior_incorrect_d)[1],
+        _calculate_recall(oracle_true_positives, oracle_false_negatives),
+        _calculate_precision(oracle_true_positives, oracle_false_positives),
+        calculate_acc(oracle_correct_d, oracle_incorrect_d)[1],
+    )
+
+
+def _offset(start, end):
+    return "{}_{}".format(start, end)
+
+
+def calculate_acc(correct_by_label, incorrect_by_label):
+    acc_by_label = dict()
+    all_keys = set(correct_by_label.keys()).union(set(incorrect_by_label.keys()))
+
+    total_correct = sum(correct_by_label.values())
+    total_incorrect = sum(incorrect_by_label.values())
+    acc = _calculate_precision(total_correct, total_incorrect)
+
+    for label in sorted(all_keys):
+        correct = correct_by_label.get(label, 0)
+        incorrect = incorrect_by_label.get(label, 0)
+        acc_by_label[label] = _calculate_precision(correct, incorrect)
+    return acc, acc_by_label
+
+
+def _calculate_precision(true_positives: int, false_positives: int):
+    if true_positives == 0:
```

Comment:
Also here I'd remove the typing, because it's not really used anywhere else. But hopefully at some point we'll use typing everywhere ;-)